Classes/Event/EventPublisher.php

```diff
@@ -105,7 +106,10 @@ public function publishMany(string $streamName, array $events, int $expectedVers
         foreach ($rawEvents as $rawEvent) {
             $eventClassName = $this->eventTypeResolver->getEventClassNameByType($rawEvent->getType());
             $event = $this->propertyMapper->convert($rawEvent->getPayload(), $eventClassName, $configuration);
-            foreach ($this->eventListenerLocator->getSynchronousListeners($rawEvent->getType()) as $listener) {
+            foreach ($this->eventListenerLocator->getSynchronousListenersByEventType($rawEvent->getType()) as $listener) {
+                if ($listener[0] instanceof ActsBeforeInvokingEventListenerMethodsInterface) {
+                    $listener[0]->beforeInvokingEventListenerMethod($listener[1], $event, $rawEvent);
```

Comment:
> you (we all) are using it as an array elsewhere already

No, we only use it as `call_user_func($listener, $event, $rawEvent);` which supports all `\callable`s

> So should we check if it's an array and if it's not, ignore it in the relevant parts?

Even if it's an array it could be in the form `['<ClassName>', '::<MethodName>']`..

> Actually, I think it's YAGNI

Probably. The thinking behind it was: There should be only one place where a listener is resolved. and the common "interface" for a listener would be something that you can just invoke. Maybe we can find a better way than `\callable`

> As far as I can see it doesn't have an impact on userland anyway, right?

Right, maybe add a comment anyways in case we change this at some point?